src/lab1/structures/camera/Camera.ts

```diff
@@ -1,18 +1,18 @@
+import { Rotatable, Translatable } from '../../../lab3/types/Transformable';
 import Vector3D from '../vector/Vector3D';
 import Vertex3D from '../vertex/Vertex3D';
 
-export default class Camera {
-  public readonly rightVector: Vector3D;
-  public readonly upVector: Vector3D;
+export default class Camera implements Translatable, Rotatable {
+  public rightVector: Vector3D;
+  public upVector: Vector3D;
 
   constructor(
-    public readonly focalPoint: Vertex3D,
-    public readonly viewVector: Vector3D,
-    // like 16:9
-    public readonly widthToHeightRatio: number,
+    public focalPoint: Vertex3D,
+    public viewVector: Vector3D,
     public readonly hFieldOfViewRads: number,
     // number of horizontal pixels
-    public readonly hResolution: number
+    public readonly hResolution: number,
+    public readonly vResolution: number
```

Comment:
Naming is not quite clear. It should be `horizontalResolution` and `verticalResolution`. Or you can name this variables as `xResolution` and `yResolution`